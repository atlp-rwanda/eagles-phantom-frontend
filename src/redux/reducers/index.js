import { combineReducers } from 'redux';
import loginReducer from './loginReducer';
import logoutReducer from './reducerLogout'

const allReducers = combineReducers({
    login: loginReducer,
    logout: logoutReducer
});


export default allReducers;

