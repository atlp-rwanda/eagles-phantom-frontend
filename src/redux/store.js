import {createStore, applyMiddleware } from 'redux';
import { composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import allReducers from './reducers/';

// const rootReducer = combineReducers({
//   count: counterReducer,
// });

// const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const middleware = applyMiddleware(thunk, promise)
const store = createStore(allReducers, {}, composeWithDevTools(middleware));

export default store;
