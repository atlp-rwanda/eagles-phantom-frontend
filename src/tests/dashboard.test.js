import React from 'react';
import Dashboard from '../components/loginPage/dashboard';
import {render } from '@testing-library/react'
import { Provider } from 'react-redux';
import {store} from '../redux/store';

const DashboardComponent = () => {
    return render(
            <Dashboard/>
    )
}

describe('test dashboard page', () =>{
    it('should render the dashboard component', () => {
        const { asFragment } = DashboardComponent();
        expect(asFragment(<Dashboard/>)).toMatchSnapshot();
    })
})