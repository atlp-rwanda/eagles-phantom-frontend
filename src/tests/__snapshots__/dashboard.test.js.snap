// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test dashboard page should render the dashboard component 1`] = `
<DocumentFragment>
  <div
    class="dashboard"
  >
    <h1>
      You are not logged in, Please Sign in first.
    </h1>
  </div>
</DocumentFragment>
`;
